src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -171,32 +171,41 @@ private TypedExpression CreateControlCall(BuiltIn opInfo, IEnumerable<OpProperty
                 /// <summary>
                 /// Creates an operation call from the conditional control API for non-literal Result comparisons.
                 /// </summary>
-                private TypedExpression CreateApplyConditionallyExpression(TypedExpression conditionExpr1, TypedExpression conditionExpr2, QsScope conditionScope, QsScope defaultScope)
+                private TypedExpression CreateApplyConditionallyExpression(TypedExpression conditionExpr1, TypedExpression conditionExpr2, QsScope equalityScope, QsScope inequalityScope)
                 {
-                    var (isConditionValid, conditionId, conditionArgs) = IsValidScope(conditionScope);
-                    var (isDefaultValid, defaultId, defaultArgs) = IsValidScope(defaultScope);
+                    if (equalityScope == null && inequalityScope == null)
+                    {
+                        return null; // Not sure how this would happen
```

Comment:
What happens if you return null? Is the if-statement just left unchanged? 
Since this shouldn't happen, a QsCompilerDiagnostic would probably make sense here.

Reply:
When it goes to create the surrounding statement, in CreateControlStatement, it checks if the created expression is null. If so, it just returns the original statement.

Reply:
The comment was in reference to the condition. I am not sure how the condition would be met. I don't know how we could have an if-statement whose bodies are all null.